R/programming.R

```diff
@@ -50,7 +53,7 @@ substitute2 = function(expr, env) {
     # null is fine, will be escaped few lines below
   } else if (is.environment(env)) {
     env = as.list(env, all.names=TRUE, sorted=TRUE)
-  } else if (!is.list(env)) {
+  } else if (!only.list(env) && !(is.AsIs(env) && only.list(rm.AsIs(env)))) {
```

Comment:
is this part about `AsIs` being tested? Not clear from a glance at the tests

Reply:
Some tests failed without this part here